### ISSUE TITLE
How to call uow.commit() in the same transaction?

### ISSUE THREAD

**Issue by AllanOricil** [apex-enterprise-patterns/fflib-apex-common#371]
I have a feature which I have to do an Update after an Insertion, on the same Transaction. 

Basically I insert a Record with Status = 'New'
then I relate a bunch of other Objects to this Record
then I finally need to update this Record Status to 'Publish'

I was previously doing this Update in a Batch, but I must remove this batch and do it in the same transaction as I don't  care if it will work or not. 

I tried the code below and it didnt work

```java
fflib_ISObjectUnitOfWork uow = DSP_Application.UnitOfWork.newInstance();

MYCUSTOMOBJECT c = new MYCUSTOMOBJECT(Status__c = 'New');
uow.registerNew(c);
...
uow.commit();


uow.registerDirty(new MYCUSTOMOBJECT(Status__c = 'Publish')); 
uow.commit();
````

then I tried and it also did not work

````java
fflib_ISObjectUnitOfWork uow1 = DSP_Application.UnitOfWork.newInstance();

MYCUSTOMOBJECT c = new MYCUSTOMOBJECT(Status__c = 'New');
uow1.registerNew(c);
...
uow1.commit();

fflib_ISObjectUnitOfWork uow2 = DSP_Application.UnitOfWork.newInstance();
uow2.registerDirty(new MYCUSTOMOBJECT(Status__c = 'Publish')); 
uow2.commit();
````

So, what is the way of doing two commits in the same Transaction?

**Comment by wimvelzeboer** [comment]
@AllanOricil the IDoWork is run at the very end of the UnitOfWork, so when all the insert & update operations (including the trigger logic) has been executed. 

So, the implementation of IDoWork should work for you.

**Comment by wimvelzeboer** [comment]
@AllanOricil Yes, that would make the most sense to add the IDoWork implementation as a private sub-class.

**Comment by cropredyHelix** [comment]
Looking at your original question, I would do the following (and changing to use standard objects)

    Account a = new Acount(Name = 'foo');
    uow.registerNew(a);

    Contact c = new Contact(lastName = 'bar')
    uow.registerNew(c,Contact.AccountId,a);
   
    a.Status__c = 'updated';
    uow.commitWork();

If you have the object references available, you can update the values at any point before the `commitWork()`

Otherwise, the two {uow object : commitWork()} pairs is perfectly fine as it resolves to effectively this:

    insert a;
    // set c.AccountId = a.Id;
    insert c;
    update a;

all in one transaction
